master/internal/rm/kubernetesrm/informer.go

```diff
@@ -8,70 +8,43 @@ import (
 	"k8s.io/client-go/tools/cache"
 	watchtools "k8s.io/client-go/tools/watch"
 
-	"github.com/determined-ai/determined/master/pkg/actor/actors"
+	"github.com/sirupsen/logrus"
 
 	k8sV1 "k8s.io/api/core/v1"
 	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
-
-	"github.com/determined-ai/determined/master/pkg/actor"
 )
 
-const defaultInformerBackoff = 5 * time.Second
-
-// messages that are sent to the informer.
-type (
-	startInformer struct{}
-)
+const defaultBackoff = 5 * time.Second
 
-// messages that are sent by the informer.
-type (
-	podStatusUpdate struct {
-		updatedPod *k8sV1.Pod
-	}
-)
+type podCallbackFunc func(*k8sV1.Pod)
 
 type informer struct {
+	syslog       *logrus.Entry
 	podInterface typedV1.PodInterface
-	podsHandler  *actor.Ref
+	podHandler   podCallbackFunc
 }
 
 func newInformer(
+	namespace string,
 	podInterface typedV1.PodInterface,
-	podsHandler *actor.Ref,
+	podHandler func(*k8sV1.Pod),
 ) *informer {
 	return &informer{
+		syslog:       logrus.WithField("Informer", namespace),
 		podInterface: podInterface,
-		podsHandler:  podsHandler,
-	}
-}
-
-// Receive implements the actor interface.
-func (i *informer) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		ctx.Tell(ctx.Self(), startInformer{})
-
-	case startInformer:
-		i.startInformer(ctx)
-
-	case actor.PostStop:
-
-	default:
-		ctx.Log().Errorf("unexpected message %T", msg)
-		return actor.ErrUnexpectedMessage(ctx)
+		podHandler:   podHandler,
 	}
-
-	return nil
 }
 
-func (i *informer) startInformer(ctx *actor.Context) {
+// startInformer returns the updated pod, if any.
+func (i *informer) startInformer() bool {
```

Comment:
Seems like a good opportunity to have this accept a `ctx context.Context` so you can pass that into stuff like `i.podInterface.List` and `i.podInterface.Watch`.

Reply:
good point